### PR TITLE
Implement aten::upsample_linear1d on mps

### PR DESCRIPTION
**Pull request by kaieberl**
Related to #77764 

cc @kulinseth @albanD @malfet @DenisVieriu97 @razarmehr

### CODE REVIEW

**Review by malfet — CHANGES_REQUESTED** [review]
Hmm, you implement linear as an alias to bilinear, which is probably acceptable, but then why not do a simple redispatch + warning (because bilinear != linear)?

File: aten/src/ATen/native/mps/operations/UpSample.mm [hunk 1]
```diff
@@ -361,6 +365,33 @@ static bool check_mps_compatibility(const c10::string_view resize_mode_str, c10:
   }
 }
 
+TORCH_IMPL_FUNC(upsample_linear1d_out_mps)
+(const Tensor& input,
+ IntArrayRef output_size,
+ bool align_corners,
+ c10::optional<double> scale,
```
[the remark this answers is not in the document]
Comment (kaieberl) [reply aten/src/ATen/native/mps/operations/UpSample.mm:372]: Sorry I missed the part with c10::optional not supported, will change this

[the remark this answers is not in the document]
Comment (kaieberl) [reply aten/src/ATen/native/mps/operations/UpSample.mm]: I replaced `c10::optional` now.
Here is an explanation on the bilinear vs linear topic: https://en.wikipedia.org/wiki/Bilinear_interpolation

> As the name suggests, the bilinear interpolant is not linear; but it is linear (i.e. affine) along lines parallel to either the x or the y direction, equivalently if x or y is held constant. Along any other straight line, the interpolant is quadratic. Even though the interpolation is not linear in the position (x and y), at a fixed point it is linear in the interpolation values, as can be seen in the (matrix) equations above.

In this case, the second dimension is constant and thus the interpolation is equivalent to the linear case.

**Review by malfet — DISMISSED** [review]
Thank you for the explanation and the updates. Looks good to me.

**Comment by kaieberl** [comment]
@malfet Is there something else I need to do?

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**